Source/Fuse.Controls.Video/Video.ScriptClass.uno

```diff
@@ -0,0 +1,91 @@
+using Uno;
+using Uno.UX;
+
+using Fuse.Scripting;
+
+namespace Fuse.Controls
+{
+	public partial class Video
+	{
+		static Video()
+		{
+			ScriptClass.Register(typeof(Video),
+				new ScriptMethod<Video>("getPosition", getPosition, ExecutionThread.Any),
+				new ScriptMethod<Video>("getDuration", getDuration, ExecutionThread.Any),
+				new ScriptMethod<Video>("resume", resume, ExecutionThread.MainThread),
+				new ScriptMethod<Video>("pause", pause, ExecutionThread.MainThread),
+				new ScriptMethod<Video>("stop", stop, ExecutionThread.MainThread));
+		}
+
+		/**
+			Gets the current position in the Video in seconds.
+
+			@scriptmethod getPosition()
+		*/
+		static object getPosition(Context c, Video v, object[] args)
+		{
+			if (args.Length != 0)
+				Fuse.Diagnostics.UserError("getPosition takes 0 arguments, but " + args.Length + " was supplied", v);
+
+			var ret = 0.0;
+			lock (v)
+				ret = v._outPosition;
+			return (object)ret;
```

Comment:
why not the simpler
```uno
lock (v)
	return (object)v._outPosition;
```
instead?

Also, [`lock (this)` is generally frowned upon in C#](https://stackoverflow.com/questions/251391/why-is-lockthis-bad). I don't think this case is particularly bad, but perhaps you could add a `object _positionMutex = new object();`  just to prevent problem if someone else tries to lock `Fuse.Controls.Video`?

Reply:
should i add this to Video? or in getPosition?